app/models/task.rb

```diff
@@ -0,0 +1,37 @@
+# == Schema Information
+#
+# Table name: tasks
+#
+#  id          :integer          not null, primary key
+#  name        :string
+#  description :text
+#  state       :integer          default(-1)
+#  project_id  :integer
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+# Indexes
+#
+#  index_tasks_on_name  (name) UNIQUE
+#
+
+class Task < ActiveRecord::Base
+  belongs_to :project
+
+  validates :name, presence: true
```

Comment:
I noticed in your migration that you made the `name` field with a unique index. However you don't validate it as well. It would be a good idea to do that rather than have the db catch the error and throw something harder than you would probably like.